src/DataTypes/DataTypeTuple.cpp

```diff
@@ -98,21 +98,38 @@ std::string DataTypeTuple::doGetPrettyName(size_t indent) const
 {
     size_t size = elems.size();
     WriteBufferFromOwnString s;
-    s << "Tuple(\n";
 
-    for (size_t i = 0; i != size; ++i)
+    /// If the Tuple is named, we will output it in multiple lines with indentation.
+    if (have_explicit_names)
     {
-        if (i != 0)
-            s << ",\n";
+        s << "Tuple(\n";
 
-        s << fourSpaceIndent(indent + 1);
-        if (have_explicit_names)
-            s << backQuoteIfNeed(names[i]) << ' ';
+        for (size_t i = 0; i != size; ++i)
+        {
+            if (i != 0)
+                s << ",\n";
+
+            s << fourSpaceIndent(indent + 1)
+                << backQuoteIfNeed(names[i]) << ' '
+                << elems[i]->getPrettyName(indent + 1);
+        }
+
+        s << ')';
```

Comment:
Just noticed, why we don't print ')' on a separate line with indentation?
Is this:
```
Tuple(
    a UInt32,
    b Tuple(
        c UInt32,
        d String))
```

Better than this:
```
Tuple(
    a UInt32,
    b Tuple(
        c UInt32,
        d String
    )
)
```

Reply:
No particular reason - I was formatting some examples, and found that this variant was more pleasant - but it might be a random preference.